src/pypdfium2/_helpers/page.py

```diff
@@ -302,7 +305,7 @@ def get_objects(self, filter=None, max_depth=15, form=None, level=0):
                 raise PdfiumError("Failed to get pageobject.")
             
             # Don't register as child object, because the lifetime of pageobjects that are part of a page is managed by pdfium. The parent page should remain alive while a pageobject is used, but it seems unjustified to store countless of weakrefs just to lock pageobjects when the parent page is closed.
-            helper_obj = PdfObject(raw_obj, page=self, pdf=self.pdf, container=form, level=level)
+            helper_obj = PdfObject(raw_obj, page=self, pdf=self.pdf, container=form, level=level, textpage=textpage)
```

Comment:
I wonder what's the best solution here. Perhaps add a `textpage` argument to `PdfObject` that is ignored but can be picked up by subclass?
An alternative might be checking `helper_obj.type == pdfium_c.FPDF_PAGEOBJ_TEXT` and post-assigning the textpage in that case, but I'm -1 on pulling in type if-checks here.